ngraph/src/ngraph/op/non_max_suppression.cpp

```diff
@@ -477,3 +419,110 @@ namespace ngraph
+}
+
+op::v4::NonMaxSuppression::NonMaxSuppression(
+    const Output<Node>& boxes,
+    const Output<Node>& scores,
+    const op::v4::NonMaxSuppression::BoxEncodingType box_encoding,
+    const bool sort_result_descending,
+    const element::Type& output_type)
+    : op::v3::NonMaxSuppression({boxes,
+                                 scores,
+                                 op::Constant::create(element::i64, Shape{}, {0}),
+                                 op::Constant::create(element::f32, Shape{}, {.0f}),
+                                 op::Constant::create(element::f32, Shape{}, {.0f}),
+                                 box_encoding,
+                                 sort_result_descending,
+                                 output_type})
+{
+    constructor_validate_and_infer_types();
+}
+
+shared_ptr<Node>
+    op::v4::NonMaxSuppression::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    NODE_VALIDATION_CHECK(this,
+                          new_args.size() >= 2 && new_args.size() <= 5,
+                          "Number of inputs must be 2, 3, 4 or 5");
+    if (new_args.size() == 5)
+    {
+        return make_shared<op::v4::NonMaxSuppression>(new_args.at(0),
+                                                      new_args.at(1),
+                                                      new_args.at(2),
+                                                      new_args.at(3),
+                                                      new_args.at(4),
+                                                      m_box_encoding,
+                                                      m_sort_result_descending,
+                                                      m_output_type);
+    }
+    else if (new_args.size() == 4)
+    {
+        return make_shared<op::v4::NonMaxSuppression>(
+            new_args.at(0),
+            new_args.at(1),
+            new_args.at(2),
+            new_args.at(3),
+            op::Constant::create(element::f32, Shape{}, {.0f}),
+            m_box_encoding,
+            m_sort_result_descending,
+            m_output_type);
+    }
+    else if (new_args.size() == 3)
+    {
+        return make_shared<op::v4::NonMaxSuppression>(
+            new_args.at(0),
+            new_args.at(1),
+            new_args.at(2),
+            op::Constant::create(element::f32, Shape{}, {.0f}),
+            op::Constant::create(element::f32, Shape{}, {.0f}),
+            m_box_encoding,
+            m_sort_result_descending);
+    }
+    else
+    {
+        return make_shared<op::v4::NonMaxSuppression>(
+            new_args.at(0),
+            new_args.at(1),
+            op::Constant::create(element::i32, Shape{}, {0}),
+            op::Constant::create(element::f32, Shape{}, {.0f}),
+            op::Constant::create(element::f32, Shape{}, {.0f}),
+            m_box_encoding,
+            m_sort_result_descending,
+            m_output_type);
+    }
```

Comment:
Nitpick: I think the same patch to avoid several big branches can be applied here